### DISCUSSION TITLE
Help to find instructions set for TJC3224T132 LCD(Ender 3 V3 SE)

### DISCUSSION THREAD

**Discussion by navaismo** [mriscoc/Ender3V2S1#1374] [Q&A]
Hello,
I found this really cool project that can handle several kinds of LCD for 3D printers with Marlin, I'm becoming crazy trying to find the low level instruction set for the LCD TJC3224T132 that Creality uses in the Ender 3 V3 SE printer.

I've modified the firmware of this printer [to show details when a print job starts from octoprint Server](https://github.com/navaismo/Ender-3V3-SE/tree/OctoPrintDetailsPageinLCD). Now I want  to see the Thumbnail of the GCODE Preview in the LCD, [according the code it should be possible to write to the SRAM and render from it the Image](https://github.com/navaismo/Ender-3V3-SE/blob/5d6144202454833e0e380daa19494f7dd1348de7/Marlin/src/lcd/dwin/dwin_lcd.cpp#L653). But so far my LCD is not responding to the instructions.

I've dug deep in the web to find info related to the LCD(with not much luck) based on the code and found this [DWIN datasheet that has almost the same intructions set as in the Ender's code](https://ecdn6.globalso.com/upload/p/1355/source/2024-07/T5L_TA-Development-GuideV3.pdf) but nor the SRAM writing or the QR instructions are working for my LCD model.

If any of you have any info related or can point me into the right direction would be great and much appreciated even if its not for free, I'm really now obsessed with this chinese LCD to let me write my own images into it.

Image of the LCD:
![TJC3224T1132_011N](https://github.com/user-attachments/assets/58650b68-0ab4-49ec-bad8-8c970cee2706)

**Comment by eduard-sukharev** [comment]
Hey there. Have you seen this documented attempt to reverse-engineer the screen via UART?

**Reply by navaismo** [reply]
Hello! No I haven't can you share a link?
I was thinking the same to put a "thing in the middle" to catch the instrcutions but not sure if that will give me acces to the SRAM.

**Reply by eduard-sukharev** [reply]
Sorry, missed the link in the original comment
https://jpcurti.github.io/E3V3SE_display_klipper/tjc3224_reverse_engineering.html
there's also his thread on Reddit
https://www.reddit.com/r/Ender3V3SE/comments/194dyur/its_time_to_make_klipper_work_with_the_original/

**Reply by navaismo** [reply]
Ah very cool, they found the same doc as I did and same conclusion not all instructions are the same unfortunately.
I saw that the printer can render a QR code when an error arise, so probably the QR instruction is another one. I guess its time to use this kind of approach too to find more instructions.

What a pitty that the real manufacturer has totally different instruction set than creality beacuse there is no way to implement our own GUI or upload images on demand.

Thanks a lot for the heads up.

**Comment by mriscoc** [comment]
If you compile the sources from here: https://github.com/CrealityOfficial/Ender-3V3-SE and flash your printer, it works ok?

**Reply by navaismo** [reply]
Hi, yes that's were my fork come from. But as the link shared above by Eduard not all instructions works, the important ones to write into SRAM or QR code generation. 

It's odd how this LCD works.

**Reply by eduard-sukharev** [reply]
how hard would it be to brute-force the qr code and sram commands?

**Reply by navaismo** [reply]
I guess will be time consuming as all brute force attacks, my approach will be:

1. Get the RS232 adapater to put it in the middle.
2. Force the printer to render a QR error.
3. Check the Command and try to duplicate with custom string.
4. From there another look into Internet to find similar command set and check for SRAM command.
5. If nothing found then disconnect LCD from printer and connect to PC to start flooding commands till something renders xD

**Reply by eduard-sukharev** [reply]
regarding the QR-code rendering - it's simply an icon. Calling
```cpp
    if (Chinese == HMI_flag.language)
      DWIN_ICON_Not_Filter_Show(ICON, ICON_Level_ERR_QR_CH, ICON_LEVEL_ERR_X, ICON_LEVEL_ERR_Y);
    else
      DWIN_ICON_Not_Filter_Show(ICON, ICON_Level_ERR_QR_EN, ICON_LEVEL_ERR_X, ICON_LEVEL_ERR_Y);
```
renders one of the two hardcoded QR-codes. As simple as that.

**Reply by navaismo** [reply]
Ah too bad I was looking to make work the serial command:

AA 96 00 64 00 64 04 68 74 74 70 3A 2F 2F 77 77 77 2E 64 77 69 6E 2E636F6D2E636E 2F CC 33 C3 3C

Still looking into it, when I have some spare time at least for the SRAM.

**Reply by eduard-sukharev** [reply]
Whoa, I totally missed that! Gotta try that thing too :D
